src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -892,17 +924,38 @@ private StableReverseOrderListView(List<E> base) {
             // This method does not evaluate the elements
             @Override
             public String toString() {
-                final StableValueImpl<E>[] delegates = ((StableList<E>)base).delegates;
+                final StableList<E> deepRoot = deepRoot(base);
+                final StableValueImpl<E>[] delegates = deepRoot.delegates;
+                // Todo: Provide a copy free solution
                 final StableValueImpl<E>[] reversed = ArraysSupport.reverse(
                         Arrays.copyOf(delegates, delegates.length));
-                return StableUtil.renderElements(base, "Collection", reversed);
+                return StableUtil.renderElements(this, "Collection", reversed);
```

Comment:
```suggestion
                final StableValueImpl<E>[] delegates = deepRoot(base).delegates;
                // Todo: Provide a copy free solution
                return StableUtil.renderElements(this, "Collection", ArraysSupport.reverse(Arrays.copyOf(delegates, delegates.length)));
```

Same as above, simplifying the code by removing local variables that are only used once